rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -46,6 +46,17 @@ def read(specification)
       @repo_reader.read(specification)
     end
 
+    def update_messages(messages)
+      hashes = messages.map(&:to_h)
+      hashes.each{|h| h[:id] = h.delete(:event_id) }
+      for_update = messages.map(&:event_id).to_set
+      start_transaction do
+        existing = Event.where(id: for_update).pluck(:id).to_set
+        (for_update - existing).each{|id| raise RubyEventStore::EventNotFound.new(id) }
```

Comment:
Maybe `tap` on `try(:first)` instead of `each`? I'm curious whether mutant will complain on that `each` :)

Reply:
This is _easier_ imho to do nothing for empty case compared to try/if statements.